internal/host/host.go

```diff
@@ -1412,3 +1418,99 @@ func (m *Manager) HandleReclaimBootArtifactDownload(ctx context.Context, h *mode
 func (m *Manager) HandleReclaimFailure(ctx context.Context, h *models.Host) error {
 	return m.sm.Run(TransitionTypeReclaimFailed, newStateHost(h), &TransitionArgsUnbindHost{ctx: ctx, db: m.db})
 }
+
+// autorenameHost renames a host if it is necessary and possible. For example, if the name is
+// localhost.localdomain and the MAC address of the main network interface is A8:CD:16:AE:79:01 it
+// will be renamed to a8-cd-16-ae-79-01. Returns a boolean indicating if the host was actually
+// renamed.
+func (m *Manager) autorenameHost(host *models.Host, inventory *models.Inventory) bool {
+	if host.RequestedHostname != "" {
+		return false
+	}
+	if inventory == nil || inventory.Hostname == "" {
+		return false
+	}
+	if !funk.Contains(hostutil.ForbiddenHostnames, inventory.Hostname) {
+		return false
+	}
+	if len(inventory.Interfaces) < 1 {
+		return false
+	}
+	mainNIC, err := m.mainNIC(inventory)
+	if err != nil {
+		m.log.WithError(err).WithFields(logrus.Fields{
+			"host": host.ID,
+			"name": inventory.Hostname,
+		}).Error(
+			"Host hasn't been automatically renamed because an unexpected error " +
+				"ocurred while trying to find the main network interface card",
+		)
+		return false
+	}
+	if mainNIC == nil {
+		m.log.WithFields(logrus.Fields{
+			"host": host.ID,
+			"name": inventory.Hostname,
+		}).Info(
+			"Host hasn't been automatically renamed because there is no network " +
+				"interface card with a MAC address that can be used to calculate " +
+				"the new name",
+		)
+		return false
+	}
+	host.RequestedHostname = strings.ToLower(strings.ReplaceAll(mainNIC.MacAddress, ":", "-"))
+	m.log.WithFields(logrus.Fields{
+		"host": host.ID,
+		"old":  inventory.Hostname,
+		"new":  host.RequestedHostname,
+		"nic":  mainNIC.Name,
+		"mac":  mainNIC.MacAddress,
+	}).Info("Host has been automatically renamed")
+	return true
+}
+
+// mainNIC returns the main network interface of the given host inventory, or nil if there is no
+// such network interface card. The main network interface card is the first that has assigned both
+// a MAC address and a global IP address.
+func (m *Manager) mainNIC(inventory *models.Inventory) (result *models.Interface, err error) {
+	for _, candidate := range inventory.Interfaces {
+		hasMAC := candidate.MacAddress != ""
+		hasV4 := false
+		for _, ip := range candidate.IPV4Addresses {
+			hasV4, err = m.isGlobal(ip)
+			if err != nil {
+				return
+			}
+			if hasV4 {
+				break
+			}
+		}
+		hasV6 := false
+		for _, ip := range candidate.IPV6Addresses {
+			hasV6, err = m.isGlobal(ip)
+			if err != nil {
+				return
+			}
+			if hasV6 {
+				break
+			}
+		}
+		if hasMAC && (hasV4 || hasV6) {
+			result = candidate
+			return
+		}
+	}
+	return
+}
+
+// isGlobal determines if the given IP address is a global one and not a local one like 127.0.0.1
+// or ::1. Returns an error if the IP address can't be parsed.
+func (m *Manager) isGlobal(ip string) (result bool, err error) {
```

Comment:
Just a nit regarding naming: I would rather have the function name a more verbose, because currently we are in the `host` package and look at `isGlobal` function. Anyone seeing it outside of where it's implemented will be confused about it's meaning, i.e. seeing a function `host.isGlobal("some string")` it's hard to guess it's about IP address.

Apart from the above, I would think whether we could move this function to `internal/network/utils.go` or `internal/network/family_utils.go` as those files already have bunch of helper functions related to networking. Apart from this, I can see like such a function could be reused in other places so having it as a public utility function is useful (we are doing "is address non-local" validation in other places too)

Reply:
Renamed to `IsGlobalIPAddress` and moved to `internal/network/utils.go`.